### DISCUSSION TITLE
Keep client session

### DISCUSSION THREAD

**Discussion by ghost** [pytest-dev/pytest#9258] [Q&A]
Hi, I'm trying to test some authentication and the problem I'm having is when I call `client.post` a new session is created.

For example with `requests` I can do the following

```
s = requests.Session()
s.get('https://httpbin.org/cookies/set/sessioncookie/123456789')
r = s.get('https://httpbin.org/cookies')
```

Every time I call `s.get()` the same session is used. 

I've looked at the documentation but I can't seem to figure how to replicate this when using `client`

**Comment by RonnyPfannschmidt** [comment]
as you never specified what client is, and how you use it overall, its impossible  to answer

please provide a minimal example of what you actually do, else its simply impossible to give sound advice
